var/spack/repos/builtin/packages/llvm/package.py

```diff
@@ -560,6 +562,8 @@ class Llvm(CMakePackage, CudaPackage):
     patch("add-include-for-libelf-llvm-12-14.patch", when="@12:14")
     patch("add-include-for-libelf-llvm-15.patch", when="@15")
 
+    patch("sanitizer-platform-limits-posix-xdr-macos.patch", when="@14 %apple-clang@15:")
```

Comment:
The `SANITIZER_MAC` macro had been there since very early versions and the patch is applicable starting `@10` (it's not applicable to the older versions only because the extension of the patched file used to be `.cc` instead of `.cpp` but I don't think that we should go that far). I guess, it should not hurt if we apply the patch whenever possible. I also think that the fix is not compiler- but platform-specific (it would be nice if somebody checked this by building with `%gcc`).
```suggestion
    patch("sanitizer-platform-limits-posix-xdr-macos.patch", when="@10:14 platform=darwin")
```

Reply:
Agreed